File: app/src/main/java/com/platonso/yamify/data/Favourites.java
package com.platonso.yamify.data;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.FirebaseFirestore;


public class Favourites {
    String title;
    String content;

    public Favourites() {
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public static CollectionReference getCollectionReferenceForRecipes(){
        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
        return FirebaseFirestore.getInstance().collection("favourites")
                .document(currentUser.getUid()).collection("my_favourites");
    }
}

File: app/src/main/java/com/platonso/yamify/data/Favourites.kt
package com.platonso.yamify.data

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.FirebaseFirestore

data class Favourites(
    var title: String? = null,
    var content: String? = null
) {
    companion object {
        fun getCollectionReferenceForRecipes(): CollectionReference {
            val currentUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
            return FirebaseFirestore.getInstance().collection("favourites")
                .document(currentUser?.uid ?: "").collection("my_favourites")
        }
    }
}
